### PR TITLE
[Term Entry] Swift Array Term Entry: .Filter()

### PR DESCRIPTION
**Pull request by MandyDoree**
### Description

- Added the Filter() term entry for Swift
- Closes Issue #2446 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
@MandyDoree Looks good to me as-is. But I wonder why you didn't use the tag you added to the `tags.md` file in your entry.

File: content/swift/concepts/arrays/terms/filter/filter.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+The `.filter()` method takes a single `condition` parameter. It is a [closure](https://www.codecademy.com/resources/docs/swift/closures) that returns a [boolean](https://www.codecademy.com/resources/docs/general/data-types/boolean) value indicating whether a given array element is a match.
+
+## Example
+
+```swift
+var names = ["Ariel", "Jamsin", "Cinderella", "Aurora", "Belle", "Mulan", "Tiana", "Moana"]
+
+//return all elements that start with an "A"
```
Comment (caupolicandiaz) [inline content/swift/concepts/arrays/terms/filter/filter.md:30]:
```suggestion
// Return all elements that start with an "A"
```

File: content/swift/concepts/arrays/terms/filter/filter.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+var result = names.filter( {$0.hasPrefix("A")})
+
+print(result)
+```
+
+In the example above, the `.filter()` method is searching through the `names` array. All items that meet the condition of starting with `A` will be returned.
```
Comment (caupolicandiaz) [inline content/swift/concepts/arrays/terms/filter/filter.md:36]:
```suggestion
In the example above, the `.filter()` method is used to search through the `names` array. All items that meet the condition of starting with `A` will be returned.
```

Comment (caupolicandiaz) [reply content/swift/concepts/arrays/terms/filter/filter.md:36]: I think it 's preferable to have this sentence above the code block on line 26

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@MandyDoree I've added my suggestions. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by MandyDoree** [comment]
@caupolicandiaz changes can be found in #2595. I had some issues creating the request (this is my first time), so I might have made some errors here and there that causes things to not go the way I planned it. Please let me know if anything looks wrong from your side.

I couldn't assign the issue to you, so I tagged you in it.

**Comment by caupolicandiaz** [comment]
@MandyDoree Why did you start a new PR? Did you have a problem making changes on this branch?

**Comment by MandyDoree** [comment]
@caupolicandiaz I know how to make a new PR, but I had issues figuring out how to make changes on the current branch. Those changes mainly being I could not check if I were doing it right. So I decided to go for the solution I was most certain of. And I changed the branch name to comply with the check list.

**Comment by SSwiniarski** [comment]
@MandyDoree, Closing this because of a new PR.

**Pull request closed without merging**